autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnResponseUtils.java

```diff
@@ -23,33 +21,15 @@
 
     public List<String> getYarnRecommendedDecommissionHostsForHostGroup(String clusterCrn, YarnScalingServiceV1Response yarnResponse,
             Map<String, String> hostFqdnsToInstanceId, int maxAllowedDownScale, Optional<Integer> mandatoryDownScaleCount) {
-        Set<String> consideredNodeIds = new HashSet<>();
-        Integer allowedDownscale = Math.max(maxAllowedDownScale, mandatoryDownScaleCount.orElse(0));
+        Integer allowedDownscale = mandatoryDownScaleCount.orElse(maxAllowedDownScale);
         List<String> decommissionNodes = yarnResponse.getScaleDownCandidates().orElse(List.of()).stream()
                 .map(YarnScalingServiceV1Response.DecommissionCandidate::getNodeId)
                 .map(nodeFqdn -> nodeFqdn.split(":")[0])
                 .filter(s -> hostFqdnsToInstanceId.keySet().contains(s))
                 .map(nodeFqdn -> hostFqdnsToInstanceId.get(nodeFqdn))
                 .limit(allowedDownscale)
-                .map(nodeId -> {
-                    consideredNodeIds.add(nodeId);
-                    return nodeId;
-                })
                 .collect(Collectors.toList());
 
-        //HostGroup candidates if mandatoryDownScaleCount is not met based on yarnResponse.
-        int forcedCandidatesCount = mandatoryDownScaleCount.orElse(0) - decommissionNodes.size();
-        if (forcedCandidatesCount > 0) {
-            List forcedCandidates = hostFqdnsToInstanceId.keySet().stream()
-                    .filter(nodeId -> !consideredNodeIds.contains(nodeId))
-                    .limit(forcedCandidatesCount).collect(Collectors.toList());
-
-            LOGGER.info("Forced downscaling candidates count '{}', candidates '{}' in cluster '{}' to achieve mandatory " +
-                            "downscaletarget '{}'.", forcedCandidates.size(), forcedCandidates, clusterCrn,
-                    mandatoryDownScaleCount.orElse(0));
-            decommissionNodes.addAll(forcedCandidates);
-        }
-
```

Comment:
Based on YarnBased downscaling, this code has become dead block, so removing to avoid confusion.